src/cider/exec/nextgen/context/CodegenContext.cpp

```diff
@@ -169,10 +169,26 @@ RuntimeCtxPtr CodegenContext::generateRuntimeCTX(
     runtime_ctx->addCiderSet(cider_set_desc.first);
   }
 
+  runtime_ctx->setTrimStringOperCharMaps(trim_char_maps_);
+
   runtime_ctx->instantiate(allocator);
   return runtime_ctx;
 }
 
+int CodegenContext::registerTrimStringOperCharMap(const std::string& trim_chars) {
+  if (!trim_char_maps_) {
+    trim_char_maps_ = std::make_shared<std::vector<std::vector<int8_t>>>();
+  }
+
+  auto trim_char_map = std::vector<int8_t>(256, 0);
+  // initialize map and set characters to be trimmed to 1
```

Comment:
better to use map here?

Reply:
used vector here because we could later pass `int8_t*` pointer to the runtime functions